xchange-gdax/src/main/java/info/bitrich/xchangestream/gdax/GDAXStreamingService.java

```diff
@@ -66,15 +73,27 @@ protected String getChannelNameFromMessage(JsonNode message) {
 
     @Override
     public String getSubscribeMessage(String channelName, Object... args) throws IOException {
-        GDAXWebSocketSubscriptionMessage subscribeMessage = new GDAXWebSocketSubscriptionMessage(SUBSCRIBE, product);
+        	ExchangeSpecification exchangeSpec = exchange.getExchangeSpecification();
+        GDAXWebsocketAuthData authData = null;
+        if ( exchangeSpec.getApiKey() != null ) {
+            GDAXAccountServiceRaw rawAccountService = (GDAXAccountServiceRaw) exchange.getAccountService();
+            authData = rawAccountService.getWebsocketAuthData();
```

Comment:
I would prefer not to include `getWebsocketAuthData` call into this particual method as well `xchange-stream`. User can call this by her own. Just let the user to pass websocet auth data to the xchange-stream (probably in `connect()` method?). By this you will have easier time with solving cyclic dependency issue as well as more flexible usage of the library.